src/ring_theory/polynomial/hermite.lean

```diff
@@ -125,4 +129,32 @@ end
 
 end coeff
 
+/-! ### Lemmas about `hermite n` and gaussians -/
+section gaussian
+
+lemma hermite_gauss_succ (n : ℕ) (x : ℝ) :
+  (-1 : ℝ)^(n+1) * real.exp (x^2 / 2) * (deriv^[n+1] (λ y, real.exp (-(y^2 / 2))) x)
+  = x * ((-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) -
+    deriv (λ x, (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)) x :=
+begin
+  rw [function.iterate_succ_apply', deriv_mul, deriv_const_mul,
+      (by simp [mul_comm] : deriv (λ y, real.exp (y^2 / 2)) x = x * real.exp (x^2 / 2))],
+  ring_exp,
+  { simp },
+  { simp },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ _)).1,
+    exact ((cont_diff_id.pow 2).div_const 2).neg.exp },
+end
+
+/- `hermite n` is (up to sign) the factor appearing in `deriv^[n]` of a gaussian -/
+lemma hermite_eq_deriv_gaussian : ∀ (x : ℝ) (n : ℕ),
+  aeval x (hermite n) = (-1 : ℝ)^n * real.exp (x^2 / 2) * (deriv^[n] (λ y, real.exp (-(y^2 / 2))) x)
+| _ 0       := by simp [← real.exp_add]
+| _ (n + 1) := begin
+  simp_rw [hermite_gauss_succ, hermite_succ, ← hermite_eq_deriv_gaussian],
+  simp [aeval_def, eval₂_eq_eval_map],
```

Comment:
Is this lemma useful at all?
```lean
lemma derivative_aeval {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
  (p : polynomial R) (x : A) :
  aeval x (derivative p) = p.sum (λ (n : ℕ) (a : R), (a * ↑n) • x ^ (n - 1)) :=
by simp_rw [derivative_apply, polynomial.sum, map_sum, aeval_mul, aeval_C, algebra.smul_def,
  aeval_X_pow]
```

Reply:
I think I would want more of an analog of [polynomial.deriv](https://leanprover-community.github.io/mathlib_docs/analysis/calculus/deriv.html#polynomial.deriv), such as
```
lemma polynomial.deriv_aeval {R 𝕜 : Type*} [comm_semiring R] [nontrivially_normed_field 𝕜] [algebra R 𝕜] (p : polynomial R) (x : 𝕜) :
deriv (λ (x : 𝕜), aeval x p) x = aeval x (derivative p) :=
by simp [aeval_def, eval₂_eq_eval_map]
```

Reply:
That looks like a great lemma to have to me

Reply:
If this lemma goes in `analysis.calculus.deriv.lean` then that file will need to import `polynomial.algebra_map`. Is that OK?

Reply:
Yes, I think that's fine.